torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -3653,14 +3653,23 @@ at::Tensor XLANativeFunctions::pixel_unshuffle(const at::Tensor& self,
       pixel_unshuffle)>::call(self, downscale_factor);
 }
 
-at::Tensor XLANativeFunctions::reshape_symint(const at::Tensor& self,
-                                              c10::SymIntArrayRef shape) {
+at::Tensor XLANativeFunctions::reshape(const at::Tensor& self,
+                                       at::IntArrayRef shape) {
   // See Note: [Disabling functionalization]
   if (runtime::sys_util::GetEnvBool("XLA_DISABLE_FUNCTIONALIZATION", false)) {
-    return at::native::reshape_symint(self, shape);
-  }
-  return at::functionalization::functionalize_aten_op_symint<ATEN_OP(
-      reshape)>::call(self, shape);
+    return at::native::reshape_symint(
```

Comment:
composite implicit ops should never call into the `native::` namespace as it bypasses the dispatcher.
In this case , I guess that this is why you get a Tensor that doesn't require grad: you skipped autograd and so the output never had the proper autograd graph added to it!